spirv_glsl.cpp

```diff
@@ -778,16 +772,28 @@ void CompilerGLSL::emit_header()
 		{
 			if (options.version < 310)
 				SPIRV_CROSS_THROW("At least ESSL 3.10 required for fragment shader interlock.");
-			require_extension_internal("GL_NV_fragment_shader_interlock");
+			statement("#extension GL_NV_fragment_shader_interlock : enable");
+			statement("#define SPIRV_Cross_beginInvocationInterlock() beginInvocationInterlockNV()");
+			statement("#define SPIRV_Cross_endInvocationInterlock() endInvocationInterlockNV()");
 		}
 		else
 		{
 			if (options.version < 420)
-				require_extension_internal("GL_ARB_shader_image_load_store");
-			require_extension_internal("GL_ARB_fragment_shader_interlock");
+				statement("#extension GL_ARB_shader_image_load_store : enable");
```

Comment:
This is problematic if some other code used require_extension_internal (earlier in emit_header, this is exactly the case), we'd end up with double declaration, so this should also be require_extension_internal.

Reply:
You're right, require extension may be called for GL_ARB_shader_image_load_store earlier.

Reply:
why are we even concerned with `GL_ARB_shader_image_load_store` ?